### PR TITLE
February release

### PR DESCRIPTION
**Pull request by pbiecek**

### CODE REVIEW

File: R/explain.R [hunk 1]
```diff
@@ -184,6 +192,9 @@ explain.default <- function(model, data = NULL, y = NULL, predict_function = NUL
       verbose_cat("  -> predict function  : yhat.default will be used (",color_codes$yellow_start,"default",color_codes$yellow_end,")\n", verbose = verbose)
     } else {
       verbose_cat("  -> predict function  : ",matching_yhat[1]," will be used (",color_codes$yellow_start,"default",color_codes$yellow_end,")\n", verbose = verbose)
+      if (model_info$type == "classification") {
+        verbose_cat("  -> predict function  : ",matching_yhat[1]," may treat 1st class as significant in case of multilabel target. (",color_codes$yellow_start,"default",color_codes$yellow_end,")\n", verbose = verbose)
```
Comment (pbiecek) [inline R/explain.R]: @maksymiuks does this if detect multilabel classification? 
such warning is needed only if predict function returns more than one number for one observation

File: R/model_info.R [hunk 2]
```diff
@@ -29,7 +29,7 @@
 #' aps_lm_model4 <- lm(m2.price ~., data = apartments)
 #' model_info(aps_lm_model4)
 #'
-#' model_regr_rf <- randomForest::randomForest(m2.price~., data = apartments, ntree = 50)
+#' model_regr_rf <- ranger::ranger(m2.price~., data = apartments, num.trees = 50)
```
Comment (pbiecek) [inline R/model_info.R]: in other examples there is library(ranger) function called before ranger() is called

File: R/model_info.R [hunk 3]
```diff
@@ -78,7 +84,10 @@ model_info.svm <- function(model, ...) {
 model_info.glm <- function(model, ...) {
   type <- "regression"
   package <- "stats"
-  ver <- as.character(utils::packageVersion("stats"))
+  ver <- try(as.character(utils::packageVersion(package)), silent = TRUE)
```
Comment (pbiecek) [inline R/model_info.R]: these 4 lines are repeated multiple times
according to DRY (do not repeat yourself) this shall be converted into a small function,
e.g. safe_check_pkg_version()

File: tests/testthat/helper-objects.R [hunk 4]
```diff
@@ -4,26 +4,89 @@ assign("message_prediction_breakdown", value = TRUE, envir = DALEX:::.DALEX.env)
...
-gbm2 <- gbm(survived~., data = titanic_imputed, distribution = "bernoulli")
+model_multiclassif_ranger <- ranger(status~., data = HR, num.trees = 50)
+model_multiclassif_ranger_prob <- ranger(status~., data = HR, num.trees = 50, probability = TRUE)
 
 # explain()
-p_fun_rf <- function(model, x) predict(model, x)
+p_fun_ranger <- function(model, x) predict(model, x)$predictions
 p_fun_glm <- function(model, x) predict(model, x, type = "response")
-explainer_classif_rf  <- explain(model_classif_rf, data = HR, predict_function = p_fun_rf)
+
```
Comment (pbiecek) [inline tests/testthat/helper-objects.R]: add here an information that predict.*** functions are masking the external model specific predict function

File: tests/testthat/helper-objects.R [hunk 5]
```diff
@@ -4,26 +4,89 @@ assign("message_prediction_breakdown", value = TRUE, envir = DALEX:::.DALEX.env)
...
+    pred <- list(1)
+    attr(pred, "probabilities") <- data.frame(rep(0.5, times = 100), rep(0.5, times = 100))
+  } else {
+    pred <- rep(1, times = 100)
+  }
+  pred
+}
+
+predict.gbm <- function(X.model, newdata, ...) {
+  rep(0.14, times = 100)
```
Comment (pbiecek) [inline tests/testthat/helper-objects.R]: instead of times = 100, maybe times = nrows(newdata)

File: tests/testthat/helper-objects.R [hunk 6]
```diff
@@ -4,26 +4,89 @@ assign("message_prediction_breakdown", value = TRUE, envir = DALEX:::.DALEX.env)
...
-explainer_regr_rf <- explain(model_regr_rf, data = apartments_test[1:1000, ], y = apartments_test$m2.price[1:1000])
-explainer_regr_rf_wo_y <- explain(model_regr_rf, data = apartments_test[1:1000, ])
+explainer_regr_ranger <- explain(model_regr_ranger, data = apartments_test[1:1000, ], y = apartments_test$m2.price[1:1000])
+explainer_regr_ranger_wo_y <- explain(model_regr_ranger, data = apartments_test[1:1000, ])
 explainer_regr_lm <- explain(model_regr_lm, data = apartments_test[1:1000, ], y = apartments_test$m2.price[1:1000])
-explainer_wo_data  <- explain(model_classif_rf, data = NULL)
+explainer_wo_data  <- explain(model_classif_ranger, data = NULL)
 
 
+# skip_if_no_codecov <- function() {
```
Comment (pbiecek) [inline tests/testthat/helper-objects.R]: these commented lines are not needed? of so, let's delete them

File: R/explain.R (hunk 1 above)
Comment (maksymiuks) [reply R/explain.R]: It is a bit more complicated. Cuz predict function returns exactly what we've ordered it to return. And that statement is true even if task is binary classification. We treat 1st class as significant for computations.

File: R/model_info.R (hunk 2 above)
Comment (maksymiuks) [reply R/model_info.R]: Done

File: R/model_info.R (hunk 3 above)
Comment (maksymiuks) [reply R/model_info.R]: Done

File: tests/testthat/helper-objects.R (hunk 4 above)
Comment (maksymiuks) [reply tests/testthat/helper-objects.R]: Done

File: tests/testthat/helper-objects.R (hunk 5 above)
Comment (maksymiuks) [reply tests/testthat/helper-objects.R]: Done

File: tests/testthat/helper-objects.R (hunk 6 above)
Comment (maksymiuks) [reply tests/testthat/helper-objects.R]: Done

**Comment by pbiecek** [comment]
Thanks

**Pull request merged**